File: ThalesCore/HostCommands/BuildIn/SetHSMDelay_LG.vb
''
'' This program is free software; you can redistribute it and/or modify
'' it under the terms of the GNU General Public License as published by
'' the Free Software Foundation; either version 2 of the License, or
'' (at your option) any later version.
''
'' This program is distributed in the hope that it will be useful,
'' but WITHOUT ANY WARRANTY; without even the implied warranty of
'' MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
'' GNU General Public License for more details.
''
'' You should have received a copy of the GNU General Public License
'' along with this program; if not, write to the Free Software
'' Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
'' 

Imports ThalesSim.Core.Message
Imports ThalesSim.Core.Cryptography
Imports ThalesSim.Core

Namespace HostCommands.BuildIn

    ''' <summary>
    ''' Sets an HSM response delay..
    ''' </summary>
    ''' <remarks>
    ''' This class implements the LG Racal command.
    ''' </remarks>
    <ThalesCommandCode("LG", "LH", "", "Sets an HSM response delay")> _
    Public Class SetHSMDelay_LG
        Inherits AHostCommand

        Const DELAY As String = "DELAY"

        Private _delay As String = ""

        ''' <summary>
        ''' Constructor.
        ''' </summary>
        ''' <remarks>
        ''' The constructor sets up the LG message parsing fields.
        ''' </remarks>
        Public Sub New()
            MFPC = New MessageFieldParserCollection
            MFPC.AddMessageFieldParser(New MessageFieldParser(DELAY, 3))
        End Sub

        ''' <summary>
        ''' Parses the request message.
        ''' </summary>
        ''' <remarks>
        ''' This method parses the command message. The message header and message command
        ''' code are <b>not</b> part of the message.
        ''' </remarks>
        Public Overrides Sub AcceptMessage(ByVal msg As Message.Message)
            MFPC.ParseMessage(msg)
            _delay = MFPC.GetMessageFieldByName(DELAY).FieldValue
        End Sub

        ''' <summary>
        ''' Creates the response message.
        ''' </summary>
        ''' <remarks>
        ''' This method creates the response message. The message header and message reply code
        ''' are <b>not</b> part of the message.
        ''' </remarks>
        Public Overrides Function ConstructResponse() As Message.MessageResponse
            Dim mr As New MessageResponse
            Log.Logger.MajorInfo("HSM")
            mr.AddElement(ErrorCodes._00_NO_ERROR)
            Return mr
        End Function

        ''' <summary>
        ''' Creates the response message after printer I/O is concluded.
        ''' </summary>
        ''' <remarks>
        ''' This method returns <b>Nothing</b> as no printer I/O is related with this command.
        ''' </remarks>
        Public Overrides Function ConstructResponseAfterOperationComplete() As Message.MessageResponse
            Return Nothing
        End Function
    End Class

End Namespace

File: ThalesSim.Core/Commands/Host/Implementations/SetHsmDelay_LG.cs
/*
 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2 of the License, or
 (at your option) any later version.

 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

using ThalesSim.Core.Message;
using ThalesSim.Core.Resources;

namespace ThalesSim.Core.Commands.Host.Implementations
{
    [ThalesHostCommand("LG", "LH", "Sets an HSM response delay")]
    public class SetHsmDelay_LG : AHostCommand
    {
        private string _delay;

        public SetHsmDelay_LG()
        {
            ReadXmlDefinitions();
        }

        public override void AcceptMessage(StreamMessage message)
        {
            base.AcceptMessage(message);

            if (XmlParseResult == ErrorCodes.ER_00_NO_ERROR)
            {
                _delay = KeyValues.Item("Delay");
            }
        }

        public override StreamResponse ConstructResponse()
        {
            var mr = new StreamResponse();

            Log.InfoFormat("HSM delay {0} set and ignored", _delay);

            mr.Add(ErrorCodes.ER_00_NO_ERROR);
            return mr;
        }
    }
}
